statuspage_connector.py

```diff
@@ -0,0 +1,591 @@
+            # 'skip_confirmation_notification': param.get('skip_confirmation_notification')
+        }
+
+        endpoint = 'pages/{0}/incidents/{1}/subscribers'.format(self._page_id, incident_id)  # pragma: allowlist secret
+        ret_val, response = self._make_rest_call(endpoint, action_result, json=data, method='post')
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Successfully created incident subscriber")
+
+    def _handle_list_incident_subscribers(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        incident_id = param.get('incident_id')
+        endpoint = 'pages/{0}/incidents/{1}/subscribers'.format(self._page_id, incident_id)  # pragma: allowlist secret
+
+        ret_val, response = self._make_rest_call(endpoint, action_result)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        for incident in response:
+            action_result.add_data(incident)
+
+        summary = action_result.set_summary({})
+        summary['num of subscribers'] = len(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def handle_action(self, param):
+
+        ret_val = phantom.APP_SUCCESS
+
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id: {}".format(self.get_action_identifier()))
+
+        if action_id == 'test_connectivity':
+            ret_val = self._handle_test_connectivity(param)
+        elif action_id == 'list_incidents':
+            ret_val = self._handle_list_incidents(param)
+        elif action_id == 'create_incident':
+            ret_val = self._handle_create_incident(param)
+        elif action_id == 'get_incident':
+            ret_val = self._handle_get_incident(param)
+        elif action_id == 'update_incident':
+            ret_val = self._handle_update_incident(param)
+        elif action_id == 'create_incident_subscriber':
+            ret_val = self._handle_create_incident_subscriber(param)
+        elif action_id == 'list_incident_subscribers':
+            ret_val = self._handle_list_incident_subscribers(param)
+
+        return ret_val
+
+    def initialize(self):
+
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
+        self._state = self.load_state()
+        if not isinstance(self._state, dict):
+            self.debug_print("Resetting the state file with the default format")
+            self._state = {
+                "app_version": self.get_app_json().get('app_version')
+            }
+            return self.set_status(phantom.APP_ERROR, STATUSPAGE_STATE_FILE_CORRUPT_ERR)
```

Comment:
If we are not maintaining any useful information in the state file, no need to return it in case of corrupted state file. We can continue the action flow, after resetting the state file.

Reply:
Removed return and continue the action flow, after resetting the state file.